### PR TITLE
Cleanup Publish Cache Step

### PR DESCRIPTION
**Pull request by cortinico**
Hey @redwarp 👋 

Just passing by and I'm leaving some improvements to the build configuration.
Since they're all chores, I've collected all of them in a single PR but let me know if you wish me to discard a commit or do separate PRs.

c447b97 Do not mix plugins{} with buildscript{}
0870b17 Explicitely specify the KtLint version to use
87c784f Add missing Copyright note to LICENSE
6a5f7cd Replace deprecated dependencyResolutionManagement
eba82dd Refetch the gradlew.bat
c9e9d6d Add the pre merge Github Action
6c1ce1e Extract Gradle Wrapper validation to its own Action
714ad0a Cleanup Publish Cache Step

### CODE REVIEW

File: .github/workflows/pre-merge.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Cache Gradle Folders
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/
+            ~/.gradle/wrapper/
+          key: cache-gradle-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
```
Comment (redwarp) [inline .github/workflows/pre-merge.yml:23]: Is that useful? To use the hash of the properties for the cache?
It seems to me that this file does not change often, meaning having a simple key works too. Worse case scenario, it pull the cache, that contains old stuff, and has to redownload things.
Unless it is here to ensure that said cache don't become bloated! Which could happen I guess if you start to store in the cache every single cache for every gradle versions....

File: settings.gradle.kts [hunk 2]
```diff
@@ -1,12 +1,26 @@
 rootProject.name = "gifdecoder"
 
-dependencyResolutionManagement {
```
Comment (redwarp) [inline settings.gradle.kts:3]: Hum, how could that be deprecated, as it got added in gradle 6.8? https://docs.gradle.org/6.8/release-notes.html

Comment (redwarp) [reply settings.gradle.kts:3]: See Central declaration of repositories

File: build.gradle.kts [hunk 3]
```diff
@@ -1,31 +1,32 @@
-// Top-level build file where you can add configuration options common to all sub-projects/modules.
-buildscript {
+plugins {
```
Comment (redwarp) [inline build.gradle.kts:1]: I'm out of my depth here. Would you have a link to some nice doc that explains that in more details? Thanks!

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 4]
```diff
@@ -1,6 +1,10 @@
 object Versions {
+    const val AGP = "4.1.1"
```
Comment (redwarp) [inline buildSrc/src/main/kotlin/Dependencies.kt:2]: <3

File: .github/workflows/pre-merge.yml (hunk 1 above)
Comment (cortinico) [reply .github/workflows/pre-merge.yml:23]: > Is that useful? To use the hash of the properties for the cache?

That's the file you touch every time you bump Gradle version. So it's not changing that often, but it definitely changes once you bump Gradle.

Given that a bump of Gradle requires you to download a new binary, it's reasonable to consider that a valid cache key imho :)

File: build.gradle.kts (hunk 3 above)
Comment (cortinico) [reply build.gradle.kts:1]: https://docs.gradle.org/current/userguide/plugins.html#sec:plugins_block 

Also this is often referred as Plugin DSL or the `plugins{}` block.

**Review by redwarp — APPROVED** [review]

**Pull request merged**